packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2108,19 +2110,29 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     for (_RouteEntry entry in toBeDisposed) {
       entry.dispose();
     }
-
-    switch (SchedulerBinding.instance.schedulerPhase) {
-      case SchedulerPhase.persistentCallbacks:
-        SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
-          overlay?.rearrange(_allRouteOverlayEntries);
-        });
-        break;
-      case SchedulerPhase.idle:
-      case SchedulerPhase.transientCallbacks:
-      case SchedulerPhase.midFrameMicrotasks:
-      case SchedulerPhase.postFrameCallbacks:
-        overlay?.rearrange(_allRouteOverlayEntries);
+    // Checks if any overlay entries are moved or added. Rearranges the overlay
+    // if so.
+    final List<OverlayEntry> newOverlayEntries = _allRouteOverlayEntries.toList();
+    bool requiresRearranging = newOverlayEntries.length > _lastUpdatedOverlayEntries.length;
```

Comment:
Wouldn't we require rearranging whenever `newOverlayEntries.length != _lastUpdatedOverlayEntries.length`?

Reply:
I thought we skip rearrange when there is only removal? if there is removal, newOverlayEntries.length < _lastUpdatedOverlayEntries.length